### PR TITLE
Bug/6.3/tdi 37635 fix conflict

### PR DESCRIPTION
**Pull request by wwang-talend**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**
please see the comment in https://jira.talendforge.org/browse/TDI-37635.
adjust the tdataprepdbinput component to meet the requirement of https://jira.talendforge.org/browse/TCOMP-344
https://jira.talendforge.org/browse/TCOMP-342

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:
https://jira.talendforge.org/browse/TDI-37635
https://jira.talendforge.org/browse/TCOMP-344
https://jira.talendforge.org/browse/TCOMP-342

### CODE REVIEW

**Comment by wwang-talend** [comment]
rebuild the test in jenkins as the issue is about the env, not about the current changeset.

**Comment by wwang-talend** [comment]
rebase it to master again and now some tests will fail : 
please see org.talend.components.jdbc.dataprep.JDBCInputTestIT, need to consider it. And will discuss with Seb and chenbin.

**Comment by wwang-talend** [comment]
have fixed the issue by a workaround.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/793/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/793/)

File: components-jdbc/src/main/java/org/talend/components/jdbc/datastore/JDBCDatastoreProperties.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+public class JDBCDatastoreProperties extends PropertiesImpl implements DatastoreProperties, RuntimeSettingProvider {
+
+    private transient Map<String, DBType> dyTypesInfo = new HashMap<String, DBType>();
+
+    private final static String CONFIG_FILE_lOCATION_KEY = "org.talend.component.jdbc.config.file";
+
+    public JDBCDatastoreProperties(String name) {
+        super(name);
+
+        String config_file = System.getProperty(CONFIG_FILE_lOCATION_KEY);
```
Comment (gonchar-ivan) [inline components-jdbc/src/main/java/org/talend/components/jdbc/datastore/JDBCDatastoreProperties.java]: This code for reading json config is duplicated in several classes

**Comment by gonchar-ivan** [comment]
General comments:

1. Seems, here we make special tJDBC component adoption for DataPrep product needs. 
It's not what we want in TCOMP. TCOMP idea was to have unified component implementation, which is compatible with all Talend products. If you see that something is missed and disallow you to have unified implementation you should notify about it in Components/TCOMP chat.

2. Please, write javadoc for your code. It is hard to understand code without javadoc.

**Comment by gonchar-ivan** [comment]
I've added junit transitive dep exclusion.
I'll merge after successful rebuild

**Comment by wwang-talend** [comment]
@gonchar-ivan thanks for the review.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/830/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/830/)

File: components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/di/TDataPrepDBInputDefinition.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 // 9 rue Pages 92150 Suresnes, France
 //
 // ============================================================================
-package org.talend.components.jdbc.dataprep;
+package org.talend.components.jdbc.dataprep.di;
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/di/TDataPrepDBInputDefinition.java]: it mean talend studio.

File: components-jdbc/src/main/java/org/talend/components/jdbc/datastore/JDBCDatastoreProperties.java (hunk 1 above)
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/datastore/JDBCDatastoreProperties.java]: another place may be deleted in future, so i allow the duplicated now, i will remove the duplicated code.

File: components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/di/TDataPrepDBInputProperties.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import org.talend.components.jdbc.CommonUtils;
+import org.talend.components.jdbc.RuntimeSettingProvider;
+import org.talend.components.jdbc.dataprep.DBType;
+import org.talend.components.jdbc.runtime.setting.AllSetting;
+import org.talend.daikon.properties.presentation.Form;
+import org.talend.daikon.properties.presentation.Widget;
+import org.talend.daikon.properties.property.Property;
+import org.talend.daikon.properties.property.PropertyFactory;
+
+// this component may not be necessary in future, so allow the duplicated code now.
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/di/TDataPrepDBInputProperties.java]: agree that, but this thing is not sure to do.

File: components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/JDBCInputProperties.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import org.talend.components.jdbc.dataset.JDBCDatasetDefinition;
+import org.talend.components.jdbc.dataset.JDBCDatasetProperties;
+import org.talend.components.jdbc.datastore.JDBCDatastoreProperties;
+import org.talend.components.jdbc.runtime.setting.AllSetting;
+import org.talend.daikon.properties.ReferenceProperties;
+
+public class JDBCInputProperties extends FixedConnectorsComponentProperties
+        implements IOProperties<JDBCDatasetProperties>, RuntimeSettingProvider {
+
+    public transient ReferenceProperties<JDBCDatasetProperties> dataset = new ReferenceProperties<>("dataset",
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/JDBCInputProperties.java]: this is the advice from chen bin, like you say, they may no need serialized.

File: components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/JDBCInputDefinition.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+import org.talend.components.api.component.AbstractComponentDefinition;
+import org.talend.components.api.component.ConnectorTopology;
+import org.talend.components.api.properties.ComponentProperties;
+import org.talend.components.jdbc.runtime.JDBCSource;
+import org.talend.components.jdbc.runtime.JDBCTemplate;
+import org.talend.daikon.properties.property.Property;
+import org.talend.daikon.runtime.RuntimeInfo;
+
+public class JDBCInputDefinition extends AbstractComponentDefinition {
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/dataprep/JDBCInputDefinition.java]: agree that.

File: components-jdbc/src/main/java/org/talend/components/jdbc/CommonUtils.java [hunk 6]
```diff
@@ -84,4 +89,36 @@ public static Schema getRejectSchema(ComponentProperties properties) {
 
         return null;
     }
+
+    public static Schema newSchema(Schema metadataSchema, String newSchemaName, List<Schema.Field> moreFields) {
+        Schema newSchema = Schema.createRecord(newSchemaName, metadataSchema.getDoc(), metadataSchema.getNamespace(),
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/CommonUtils.java]: i can do it.

File: components-jdbc/src/main/java/org/talend/components/jdbc/CommonUtils.java [hunk 7]
```diff
@@ -84,4 +89,36 @@ public static Schema getRejectSchema(ComponentProperties properties) {
 
         return null;
     }
+
```
Comment (wwang-talend) [inline components-jdbc/src/main/java/org/talend/components/jdbc/CommonUtils.java]: https://jira.talendforge.org/browse/TDI-37859

**Pull request merged**